From: Khem Raj <raj.khem at gmail.com>
Date: Sat, 15 Mar 2025 17:58:40 +0100
Subject: [PATCH] acinclude.m4: Add parameter names in prototype for g().

This allows it to compile with older gcc e.g. gcc-10
which does not have allow parameter name omission, it results
in

a.c: In function ‘g’:
a.c:3:8: error: parameter name omitted
    3 | void g(int,t1 const*,t1,t2,t1 const*,int){}
      |        ^~~

this was added to gcc via [1] thats why it is supported in
newer gcc.

Adding the parameter names make it compatible with
old and new gcc

[1] https://gcc.gnu.org/pipermail/gcc-cvs/2020-October/336068.html

Signed-off-by: Khem Raj <raj.khem at gmail.com>
Signed-off-by: Charlie Jenkins <charlie at rivosinc.com>
Upstream: https://gmplib.org/list-archives/gmp-devel/2025-March/006294.html
---
 acinclude.m4 | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 9cf9483..1fd0d4f 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -609,7 +609,7 @@ GMP_PROG_CC_WORKS_PART([$1], [long long reliability test 1],
 
 #if defined (__GNUC__) && ! defined (__cplusplus)
 typedef unsigned long long t1;typedef t1*t2;
-void g(){}
+void g(int a,t1 const* b,t1 c,t2 d,t1 const* e,int f){}
 void h(){}
 static __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
-- 
2.43.0